kong/plugins/jwt/jwt_parser.lua

```diff
@@ -66,6 +66,39 @@ local alg_sign = {
       return nil
     end
     return sig
+  end,
+  ES512 = function(data, key)
+    local pkey = openssl_pkey.new(key)
+    local sig = assert(pkey:sign(data, "sha512", nil, { ecdsa_use_raw = true }))
+    if not sig then
+      return nil
+    end
+    return sig
+  end,
+
+  PS256 = function(data, key)
+    local pkey = openssl_pkey.new(key)
+    local sig = assert(pkey:sign(data, "sha256", openssl_pkey.PADDINGS.RSA_PKCS1_PSS_PADDING))
+    if not sig then
+      return nil
+    end
+    return sig
+  end,
+  PS384 = function(data, key)
+    local pkey = openssl_pkey.new(key)
+    local sig = assert(pkey:sign(data, "sha384", openssl_pkey.PADDINGS.RSA_PKCS1_PSS_PADDING))
+    if not sig then
+      return nil
+    end
+    return sig
+  end,
+  PS512 = function(data, key)
+    local pkey = openssl_pkey.new(key)
+    local sig = assert(pkey:sign(data, "sha512", openssl_pkey.PADDINGS.RSA_PKCS1_PSS_PADDING))
```

Comment:
This is a log fragment of boilerplate code. Could we somehow refactor it?

Reply:
There is a follow-up task to combine this module with the openid-connect signature module, which is already refactored.